includes/frontend/class-wcmp-checkout.php

```diff
@@ -225,6 +234,8 @@ public function getDeliveryOptionsConfig(): array
             }
         }
 
+        $carrierSettings['instabox']['allowSameDayDelivery'] = $this->shouldShowSameDayDelivery();
```

Comment:
Dit is nu instabox-specifiek, maar ik zou het hier graag zo zien dat deze functionaliteit ook werkt als je deze instelling (samedaydelivery met de cutoff tijd en de fee) aan een andere carrier toevoegt, als het enigszins mogelijk is. Dus ook in de shouldShowSameDayDelivery method zou ik op een abstracter niveau kijken of dit getoond moet worden niet hardcoded alleen wanneer instabox actief is.